ada_feeding/ada_feeding/idioms/retry_call_ros_service.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module defines the retry_call_ros_service idiom, which will call a ROS
+service, optionally check the response value, and retry up to a pre-specified
+number of times if there is a failure.
+"""
+
+# Standard imports
+import logging
+from typing import Any, List, Optional
+
+# Third-party imports
+import py_trees
+from py_trees.blackboard import Blackboard
+from py_trees_ros.service_clients import FromBlackboard, FromConstant
+
+
+def retry_call_ros_service(
+    name: str,
+    service_type: Any,
+    service_name: str,
+    key_request: Optional[str] = None,
+    request: Optional[Any] = None,
+    key_response: Optional[str] = None,
+    response_checks: Optional[List[py_trees.common.ComparisonExpression]] = None,
+    max_retries: int = 3,
+    wait_for_server_timeout_sec: float = 3.0,
```

Comment:
Same here, should wait_for_server_timeout_sec be something that we have easy access to change? If not, you can ignore this comment.

Reply:
See above comment, I think changing it within the tree that calls this idiom is sufficient, I don't think we need to expose it all the way up to a parameter of the tree (so it can be changed via yaml file)